packages/component-aws-s3/src/s3_utils.ts

```diff
@@ -0,0 +1,332 @@
+import { S3Client, GetObjectCommand, PutObjectCommand, DeleteObjectCommand } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { createPresignedPost, PresignedPost } from '@aws-sdk/s3-presigned-post';
+
+import {
+    ArgOptional,
+    Communicator, CommunicatorContext,
+    ConfiguredInstance,
+    InitContext,
+    Workflow,
+    WorkflowContext
+} from '@dbos-inc/dbos-sdk';
+import { AWSServiceConfig, getAWSConfigForService, loadAWSConfigByName } from '@dbos-inc/aws-config';
+import { DBOSError } from '@dbos-inc/dbos-sdk/dist/src/error';
+
+export interface FileRecord {
+    key: string;
+}
+
+export interface S3Config{
+    awscfgname?: string,
+    awscfg?: AWSServiceConfig,
+    bucket: string,
+    s3Callbacks?: {
+        newActiveFile: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        newPendingFile: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        fileActivated: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+        fileDeleted: (ctx: WorkflowContext, rec: FileRecord) => Promise<unknown>;
+    }
+}
+
+export class S3Ops extends ConfiguredInstance {
+    //////////
+    // S3 Configuration
+    //////////
+
+    static AWS_S3_CONFIGURATION = 'aws_s3_configuration';
+
+    constructor(name: string, readonly config: S3Config) {super(name);}
+
+    async initialize(ctx: InitContext) {
+        // Get the config and call the validation
+        if (!this.config.awscfg) {
+            if (this.config.awscfgname) {
+                this.config.awscfg = loadAWSConfigByName(ctx, this.config.awscfgname);
+            }
+            else {
+                this.config.awscfg = getAWSConfigForService(ctx, S3Ops.AWS_S3_CONFIGURATION);
+            }
+        }
+        return Promise.resolve();
+    }
+
+    static createS3Client(cfg: AWSServiceConfig) {
+        return new S3Client({
+            region: cfg.region,
+            credentials: cfg.credentials,
+            maxAttempts: cfg.maxRetries,
+        });
+    }
+
+    ///////
+    //  Basic functions +
+    //  Communicator wrappers for basic functions
+    ///////
+
+    // Delete object
+    static async deleteS3Cmd(s3: S3Client, bucket: string, key: string)
+    {
+        return await s3.send(new DeleteObjectCommand({
+            Bucket: bucket,
+            Key: key,
+        }));
+    }
+
+    static async deleteS3(awscfg: AWSServiceConfig, bucket: string, key: string)
+    {
+        const s3 = S3Ops.createS3Client(awscfg);
```

Comment:
Looks like there are still several places that don't reuse the s3client -- can we replace all of them and change all static methods to instance methods?

Reply:
Those could be removed instead of replaced.  It's an option that could be used by an application but not sure if that will be a common choice, should I remove?

Reply:
Removed.